src/overlays/actors/ovl_En_Encount1/z_en_encount1.c

```diff
@@ -19,13 +26,313 @@ const ActorInit En_Encount1_InitVars = {
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 入力エラーデッス！ ☆☆☆☆☆ \n" VT_RST);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 入力エラーデッス！ ☆☆☆☆☆ \n" VT_RST);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 入力エラーデッス！ ☆☆☆☆☆ \n" VT_RST);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 入力エラーデッス！ ☆☆☆☆☆ \n" VT_RST);
+        osSyncPrintf("\n\n");
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->spawnType = (this->actor.params >> 0xB) & 0x1F;
+    this->maxCurSpawns = (this->actor.params >> 6) & 0x1F;
+    this->maxTotalSpawns = this->actor.params & 0x3F;
+    this->curNumSpawn = this->totalNumSpawn = 0;
+    spawnRange = 120.0f + (40.0f * this->actor.posRot.rot.z);
+    this->spawnRange = spawnRange;
+
+    osSyncPrintf("\n\n");
+    // It's an enemy spawner!
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 敵発生ゾーンでた！ ☆☆☆☆☆ %x\n" VT_RST, this->actor.params);
+    // Type
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 種類\t\t   ☆☆☆☆☆ %d\n" VT_RST, this->spawnType);
+    // Maximum number of simultaneous spawns
+    osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 最大同時発生数     ☆☆☆☆☆ %d\n" VT_RST, this->maxCurSpawns);
+    // Maximum number of spawns
+    osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ 最大発生数  \t   ☆☆☆☆☆ %d\n" VT_RST, this->maxTotalSpawns);
+    // Spawn check range
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発生チェック範囲   ☆☆☆☆☆ %f\n" VT_RST, this->spawnRange);
+    osSyncPrintf("\n\n");
+
+    this->actor.flags &= ~1;
+    switch (this->spawnType) {
+        case SPAWNER_LEEVER:
+            this->timer = 30;
+            this->maxCurSpawns = 5;
+            if (globalCtx->sceneNum == SCENE_SPOT13) { // Haunted Wasteland
+                this->reduceLeevers = true;
+                this->maxCurSpawns = 3;
+            }
+            this->updateFunc = EnEncount1_SpawnLeevers;
+            break;
+        case SPAWNER_TEKTITE:
+            this->maxCurSpawns = 2;
+            this->updateFunc = EnEncount1_SpawnTektites;
+            break;
+        case SPAWNER_STALCHILDREN:
+        case SPAWNER_WOLFOS:
+            if (globalCtx->sceneNum == SCENE_SPOT00) { // Hyrule Field
+                this->maxTotalSpawns = 10000;
+            }
+            this->updateFunc = EnEncount1_SpawnStalchildOrWolfos;
+            break;
+    }
+}
+
+void EnEncount1_SpawnLeevers(EnEncount1* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 floorType;
+    f32 spawnDist;
+    s32 spawnParams;
+    s16 spawnLimit;
+    s16 spawnAngle;
+    Vec3f spawnPos;
+    CollisionPoly* floorPoly;
+    s32 bgId;
+    f32 floorY;
+    EnReeba* leever;
+
+    this->outOfRangeTimer = 0;
+    spawnPos = this->actor.posRot.pos;
+
+    if ((this->timer == 0) && (globalCtx->csCtx.state == 0) && (this->curNumSpawn <= this->maxCurSpawns) &&
+        (this->curNumSpawn < 5)) {
+        floorType = func_80041D4C(&globalCtx->colCtx, player->actor.floorPoly, player->actor.floorPolySource);
+        if ((floorType != 4) && (floorType != 7) && (floorType != 12)) {
```

Comment:
Are there already floor type enums returned by func_80041D4

Reply:
I don't think so, though there probably will be once BgCheck is merged.